tests/conftest.py

```diff
@@ -5,95 +5,203 @@
     return df
 
+
 @pytest.fixture(scope="module")
 def dataframe_enc_big_na():
-    df = {'var_A': ['A'] * 6 + ['B'] * 10 + ['C'] * 4 + ['D'] * 10 + ['E'] * 2 + ['F'] * 2 + ['G'] * 6,
-          'var_B': ['A'] * 10 + ['B'] * 6 + ['C'] * 4 + ['D'] * 10 + ['E'] * 2 + ['F'] * 2 + ['G'] * 6,
-          'var_C': ['A'] * 4 + ['B'] * 6 + ['C'] * 10 + ['D'] * 10 + ['E'] * 2 + ['F'] * 2 + ['G'] * 6, }
+    df = {
+        "var_A": ["A"] * 6
+        + ["B"] * 10
+        + ["C"] * 4
+        + ["D"] * 10
+        + ["E"] * 2
+        + ["F"] * 2
+        + ["G"] * 6,
+        "var_B": ["A"] * 10
+        + ["B"] * 6
+        + ["C"] * 4
+        + ["D"] * 10
+        + ["E"] * 2
+        + ["F"] * 2
+        + ["G"] * 6,
+        "var_C": ["A"] * 4
+        + ["B"] * 6
+        + ["C"] * 10
+        + ["D"] * 10
+        + ["E"] * 2
+        + ["F"] * 2
+        + ["G"] * 6,
+    }
     df = pd.DataFrame(df)
-    df.loc[0, 'var_A'] = np.nan
+    df.loc[0, "var_A"] = np.nan
     return df
 
+
 @pytest.fixture(scope="module")
 def dataframe_normal_dist():
     np.random.seed(0)
     mu, sigma = 0, 0.1  # mean and standard deviation
     s = np.random.normal(mu, sigma, 100)
     df = pd.DataFrame(s)
-    df.columns = ['var']
+    df.columns = ["var"]
     return df
 
 
-@pytest.fixture(scope='module')
+@pytest.fixture(scope="module")
 def dataframe_constant_features():
-    data = {'Name': ['tom', 'nick', 'krish', 'jack'],
-            'City': ['London', 'Manchester', 'Liverpool', 'Bristol'],
-            'Age': [20, 21, 19, 18],
-            'Marks': [0.9, 0.8, 0.7, 0.6],
-            'dob': pd.date_range('2020-02-24', periods=4, freq='T'),
-            'const_feat_num': [1, 1, 1, 1],
-            'const_feat_cat': ['a', 'a', 'a', 'a'],
-            'quasi_feat_num': [1, 1, 1, 2],
-            'quasi_feat_cat': ['a', 'a', 'a', 'b']}
+    data = {
+        "Name": ["tom", "nick", "krish", "jack"],
+        "City": ["London", "Manchester", "Liverpool", "Bristol"],
+        "Age": [20, 21, 19, 18],
+        "Marks": [0.9, 0.8, 0.7, 0.6],
+        "dob": pd.date_range("2020-02-24", periods=4, freq="T"),
+        "const_feat_num": [1, 1, 1, 1],
+        "const_feat_cat": ["a", "a", "a", "a"],
+        "quasi_feat_num": [1, 1, 1, 2],
+        "quasi_feat_cat": ["a", "a", "a", "b"],
+    }
+
+    df = pd.DataFrame(data)
+    return df
+
+
+@pytest.fixture(scope="module")
+def dataframe_duplicate_features():
+    data = {
+        "Name": ["tom", "nick", "krish", "jack"],
+        "dob2": pd.date_range("2020-02-24", periods=4, freq="T"),
+        "City": ["London", "Manchester", "Liverpool", "Bristol"],
+        "Age": [20, 21, 19, 18],
+        "Marks": [0.9, 0.8, 0.7, 0.6],
+        "dob": pd.date_range("2020-02-24", periods=4, freq="T"),
+        "City2": ["London", "Manchester", "Liverpool", "Bristol"],
+        "dob3": pd.date_range("2020-02-24", periods=4, freq="T"),
+        "Age2": [20, 21, 19, 18],
+    }
+
+    df = pd.DataFrame(data)
+    return df
```

Comment:
Also, it seems like it is the duplicate of the last fixture. I'll delete the last fixture since tests are based on this one or maybe that got added while fixing merge, anyway I'll fix it :)